build/dev/haproxy_local_setup.sh

```diff
@@ -0,0 +1,193 @@
+if [[ "$OS" == "Darwin" ]]; then
+	# OSX uses BSD readlink
+	BASEDIR="$(dirname "$0")"
+else
+	BASEDIR=$(readlink -e "$(dirname "$0")")
+fi
+
+cd "${BASEDIR}/../.." || exit 1
+
+DC_COMMAND=$*
+
+if [[ "$DC_COMMAND" == "down" ]]; then
+  cleanup || true
+  exit
+fi
+
+echo "HAProxy base directory is ${BASEDIR}"
+# cd "${BASEDIR}"
+
+function gather_logs() {
+	docker logs http-gateway-for-testing 2> http-gateway-for-testing.log
+	docker logs core-service-for-testing 2> core-service-for-testing.log
+}
+
+function cleanup() {
+	# ----------- clean up -----------
+	echo "Stopping node roacha container"
+	docker rm -f roacha &> /dev/null || true
+
+	echo "Stopping node roachb container"
+	docker rm -f roachb &> /dev/null || true
+
+	echo "Stopping node roachc container"
+	docker rm -f roachc &> /dev/null || true
+	
+	echo "Stopping haproxy docker"
+	docker rm -f dss-crdb-cluster-for-testing &> /dev/null || true
+
+	echo "Stopping dummy oauth container"
+	docker rm -f dummy-oauth-for-testing &> /dev/null || true
+
+	echo "Stopping http gateway container"
+	docker kill -f http-gateway-for-testing &> /dev/null || true
+
+	echo "Stopping core-service container"
+	docker kill -f core-service-for-testing &> /dev/null || true
+}
+
+function on_exit() {
+	gather_logs || true
+}
+
+function on_sigint() {
+	cleanup
+	exit
+}
+
+trap on_exit   EXIT
+trap on_sigint SIGINT
+
+
+echo " -------------- BOOTSTRAP ----------------- "
+echo "Building local container for testing (see core-service-build.log for details)"
+docker build --rm . -t local-interuss-dss-image > core-service-build.log
+
+echo " ---------------- CRDB -------------------- "
+echo "cleaning up any crdb pre-existing containers"
+docker rm -f roacha &> /dev/null || echo "No CRDB to clean up"
+docker rm -f roachb &> /dev/null || echo "No CRDB to clean up"
+docker rm -f roachc &> /dev/null || echo "No CRDB to clean up"
+
+echo "Stopping haproxy container"
+	docker rm -f dss-crdb-cluster-for-testing &> /dev/null || true
+
+echo "Starting roacha with admin port on :8080"
+docker run -d --rm --name roacha \
+	-p 8080:8080 \
+	$FLAGS > /dev/null
+
+echo "Starting roachb with admin port on :8088"
+docker run -d --rm --name roachb \
+	-p 8088:8088 \
+	$FLAGS > /dev/null
+
+echo "Starting roachc with admin port on :8089"
+docker run -d --rm --name roachc \
+	-p 8089:8089 \
+	$FLAGS > /dev/null
+
+echo "Initialize cluster setup"
+docker exec -it roacha	\
+	./cockroach init --insecure
+
+echo "Waiting for cluster to initialize... "
+sleep 5
+
+echo "Generate haproxy.cfg from one of the cluster nodes"
+docker exec -it roacha ./cockroach gen haproxy --insecure
+
+
+echo "Copy haproxy.cfg file to a local folder"
+docker exec -it roacha cat haproxy.cfg > ~/Downloads/haproxy/haproxy.cfg
+
+echo "Start the HAProxy container by mounting the cfg file."
+docker run -d --name dss-crdb-cluster-for-testing --network dss_sandbox_default -p 26257:26257 -v ~/Downloads/haproxy:/usr/local/etc/haproxy:ro haproxy:1.7  
```

Comment:
atm, I am using `~/Downloads/haproxy` from local machine to mount the volume for haproxy. Let me know if I should be using something else. I tried using `/var/tmp/`, but it didn't work.